toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go

```diff
@@ -587,27 +574,25 @@ func (r *RpmRepoCloner) clonePackage(baseArgs []string, enabledRepoOrder ...stri
 }
 
 func convertPackageVersionToTdnfArg(pkgVer *pkgjson.PackageVer) (tdnfArg string) {
-	tdnfArg = pkgVer.Name
 	// TDNF does not accept versioning information on implicit provides.
 	if pkgVer.IsImplicitPackage() {
 		if pkgVer.Condition != "" {
 			logger.Log.Warnf("Discarding version constraint for implicit package: %v", pkgVer)
 		}
+		tdnfArg = pkgVer.Name
 		return
 	}
 
 	// Treat <= as =
 	// Treat > and >= as "latest"
```

Comment:
Idea: since TDNF handles version constraints better now, we might as well just add them for the >= and > cases as well. This would have the advantage of consistency, so we might avoid some developer confusion during debugging.

Also, we could then have one comment less and people wouldn't need to wonder why these constraints are special when we re-visit this function in the future.

Reply:
The logic to pick a specific version chokes very badly if we offer too many choices (since it downloads them all, then picks one). I made this change to work around a specific rpm resolution issue for the delta downloader, but when I noticed it started downloading a lot more things. So this was a concious choice to only fix what I needed. 

If we want to commit to improving the performance of the WhatProvides()... et. al. functions I'm happy to make it general (like it probably should be).